vaadin-upload-flow-parent/vaadin-upload-flow/src/main/java/com/vaadin/flow/component/upload/Upload.java

```diff
@@ -653,6 +675,37 @@ public void setReceiver(Receiver receiver) {
         } else {
             setMaxFiles(1);
         }
+
+        runBeforeClientResponse(ui -> getElement().setAttribute("target",
+                new StreamReceiver(getElement().getNode(), "upload",
+                        getStreamVariable())));
+    }
+
+    /**
+     * Set the upload handler to be used for this upload component.
+     * <p>
+     * The given handler defines how uploaded file content is handled on the
+     * server and invoked per each single file to be uploaded. Note! This method
+     * overrides the receiver set by {@link #setReceiver(Receiver)} and removes
+     * the maximum files limit.
+     *
+     * @param handler
+     *            upload handler to use for file receptions
+     */
+    public void setUploadHandler(UploadHandler handler) {
+        AbstractStreamResource elementStreamResource = new StreamResourceRegistry.ElementStreamResource(
```

Comment:
```suggestion
        ElementStreamResource elementStreamResource = new StreamResourceRegistry.ElementStreamResource(
```
We could just use the actual type instead of the abstract.